### PR TITLE
[indexer] -  module cache

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

Added module cache to indexer for parsing move objects

## Test Plan 

Added Integration test

### CODE REVIEW

**Review by 666lcz — APPROVED** [review]
LGTM

File: crates/sui-indexer/migrations/2022-11-18-195259_transactions/up.sql [hunk 1]
```diff
@@ -1,37 +1,38 @@
...
-    transaction_kinds TEXT[] NOT NULL,
+CREATE TABLE transactions
+(
+    id                          BIGSERIAL PRIMARY KEY,
+    transaction_digest          VARCHAR(255) NOT NULL,
+    sender                      VARCHAR(255) NOT NULL,
+    recipients                  TEXT[]       NOT NULL,
+    checkpoint_sequence_number  BIGINT       NOT NULL,
+    transaction_time            TIMESTAMP,
+    transaction_kind            TEXT         NOT NULL,
```
Comment (patrickkuo) [inline crates/sui-indexer/migrations/2022-11-18-195259_transactions/up.sql:9]: there was a schema mismatch causing db error

File: crates/sui-indexer/src/store/indexer_store.rs [hunk 2]
```diff
@@ -14,21 +14,21 @@ use crate::models::transactions::Transaction;
...
     fn get_checkpoint(&self, id: CheckpointId) -> Result<Checkpoint, IndexerError>;
 
-    fn get_object_with_options(
+    fn get_object(
         &self,
         object_id: ObjectID,
-        options: SuiObjectDataOptions,
-    ) -> Result<SuiObjectResponse, IndexerError>;
+        version: Option<SequenceNumber>,
+    ) -> Result<ObjectRead, IndexerError>;
```
Comment (patrickkuo) [inline crates/sui-indexer/src/store/indexer_store.rs:31]: let's try to use Sui's internal storage types in the indexer store, we can convert it to RPC types in the RPC layer, we can reuse more code from full node if we do this.

Comment (patrickkuo) [reply crates/sui-indexer/src/store/indexer_store.rs:31]: cc @gegaowp

**Pull request merged**